### DISCUSSION TITLE
Running DITA-OT 4.1.2 from Java API and extracted DITA-OT dir

### DISCUSSION THREAD

**Discussion by andreacanforadev** [dita-ot/dita-ot#4343] [Q&A]
I created a Java project and added `org.dita-ot:dost:4.1.2` as dependency and extracted DITA-OT in a directory including all source files as in github develop branch.

I am trying to run a conversion with code:

```java
ProcessorFactory pf = ProcessorFactory.newInstance(ditaDir);
pf.setBaseTempDir(new File(DATA_TEMP_DIR));

Processor p = pf.newProcessor("pdf")
		.setLogger(log)
		.setInput(inputFile)
		.setOutputDir(new File(DATA_OUPUT_DIR));

p.run();
```

I am getting exception:

```
Caused by: org.dita.dost.exception.DITAOTException: null
Caused by: org.apache.tools.ant.BuildException: import requires file attribute or at least one nested resource
```
Is there any documentation or guide on how to use Java API?

Thanks in advance!

**Comment by infotexture** [comment]
> Is there any documentation or guide on how to use Java API?

See the example in the docs at https://www.dita-ot.org/dev/reference/java-api.

**Reply by andreacanforadev** [reply]
That doesn't help much.. I am using the same code,  and it's not working

**Comment by jelovirt** [comment]
Instead of extracting the source Git repo, use the distribution build. I think the root cause here is that you haven't run the install process, and the distribution build has the install process artifacts.

**Reply by andreacanforadev** [reply]
I tried also unzipping distribution build:
Caused by: java.nio.file.NoSuchFileException: dita-ot-4.1.2/build.xml

I also tried using ditaDir = /opt/homebrew/Cellar/dita-ot/4.1.2/libexec (Mac installation dir of working dita ot)
I get different error:
Caused by: java.util.MissingResourceException: Can't find bundle for base name messages, locale en_US

**Reply by jelovirt** [reply]
Is the `ditaDir` variable in your code set to the directory that contains `build.xml`? Are the `lib` and `config` directories in classpath?
